client/src/app.js

```diff
@@ -12,26 +12,51 @@ import NoMatch from './pages/no-match';
 import Events from './pages/events';
 import Restaurant from './components/restaurant/restaurant-item';
 import Profile from './components/profile/profile';
+import Auth from './services/auth';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isAuthenticated: false,
+    };
+  }
+
+  componentDidMount() {
+    if (Auth.isAuthenticated()) {
+      this.setState({
+        isAuthenticated: true,
+      });
+    }
+  }
+
+  handleLogin() {
+    this.setState({ isAuthenticated: true });
+  }
+
   render() {
+    const { isAuthenticated } = this.state;
     return (
       <Router>
-        <NavigationBar />
+        <NavigationBar isAuthenticated={isAuthenticated} />
         <Switch>
           <Route exact path="/" component={Home} />
           <Route path="/about" component={About} />
           <Route path="/events" component={Events} />
           <Route path="/restaurants/:id" component={Restaurant} />
           <Route path="/restaurants" component={RestaurantList} />
-          <Route path="/login" component={Login} />
+          <Route
+            path="/login"
+            render={(routeProps) => {
+              return <Login loginHandler={() => { this.handleLogin(); }} {...routeProps} />;
+            }}
+          />
```

Comment:
Could you explain how lines 48-53 are working, please?

Reply:
https://reacttraining.com/react-router/web/api/Route/render-func

If user go to '/login' arrow function on line 50 return `<Login />` component with default props `routerProps` (with history, match, etc) and also I've added `loginHandler` which return `this.handleLogin()` wher `this === App`

So after user login on page `Login`  function `this.handleLogin()` will be invoked and state will be changed and all components with `isAuthenticated` will be updated

Reply:
Thanks. But as I understand, each time user goes to `/login` page, `this.state.isAuthenticated` will become true each time, won't it?

Reply:
If localStorage contains token than `isAuthenticated` will be `true` in other case - `false`

Reply:
I mean line 33 hre is invoked (and sets `isAuthenticated ` true) when user wants to go `/login` page

Reply:
> after user **login** on page Login function this.handleLogin() will be invoked

https://github.com/Social-projects-Rivne/Lunch_Time/pull/46/files#diff-77bddf7173ca118d8ca26a36e968dc3eR40

Reply:
Does `this.handleLogin()` in line 53 is invoked or passed to `<Login>` component?

Reply:
On line 51 I pass arrow function to `Login`. When user successfully login in `Login` component this arrow function will be invoked and invoke `this.handleLogin()` method.
You can add console.log after line 35 https://github.com/Social-projects-Rivne/Lunch_Time/pull/46/files#diff-77bddf7173ca118d8ca26a36e968dc3eR35 and see what object `loginHandler` is
`console.log('Login -> login -> loginHandler', loginHandler);`